### ISSUE TITLE
IIFE function expression incorrectly deparenthesized when preceded by pure annotation

### ISSUE THREAD

**Issue by mbrock** [evanw/esbuild#258]
I'm testing with esbuild 0.6.2.

The source file

```
/* @__PURE__ */ (function () {})()
```

gives the output

```
/* @__PURE__ */ function() {
}();
```

when compiled with no options. This output is not valid JavaScript: it reads as a function declaration with no name.

**Comment by mbrock** [comment]
Of the published versions, this first appears in 0.5.23 and I believe the relevant commit is https://github.com/evanw/esbuild/commit/d7679fc7dff152daed0016d1aa7771f2d55e86eb.

**Comment by mbrock** [comment]
With another kind of preserved comment, the expression stays parenthesized:

```
/* @license */
(function() {
})();
```

**Comment by mbrock** [comment]
The issue first appears in 0.5.22 when the support for pure annotations was added.

**Comment by evanw** [comment]
Thanks so much for reporting this. Makes sense why this broke. Sorry I didn't catch it. The fix is in version 0.6.3.

**Comment by mbrock** [comment]
@evanw Amazing, thanks!
